### PR TITLE
Fix NextTokenLM bug

### PR DESCRIPTION
**Pull request by leo-liuzy**

### CODE REVIEW

**Comment by epwalsh** [comment]
Looks like some of these commits are also in https://github.com/allenai/allennlp-models/pull/214, right? If we merge this first, that should be okay.

**Comment by leo-liuzy** [comment]
Ah, I think so. I deliberately submit two separate PR for different modifications; looks like it wasn't successful for some reason :(

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,12 +8,13 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ## Unreleased
 
 ### Fixed
-
+-  fixing the allennlp issue 4745.
```
Comment (epwalsh) [inline CHANGELOG.md:11]:
```suggestion

- Fixed https://github.com/allenai/allennlp/issues/4745.
```

**Review by epwalsh — APPROVED** [review]
👍

**Review by epwalsh — APPROVED** [review]

**Pull request merged**